MetriCam2/Camera.cs

```diff
@@ -1583,6 +1594,7 @@ public CameraImage CalcChannel(string channelName)
                 }
 
                 img = CalcChannelImpl(channelName);
+                imageCache.Add(channelName, img);
```

Comment:
Image is added to cache only in this branch. I would add it right before the `return` statement.